app/assets/javascripts/kernel-functions.js

```diff
@@ -111,6 +113,7 @@ function createTracesAndLayout(arr, title, jitter='all', expressionLabel){
     }
     var layout = {
         title: title,
+        xaxis: 'category',
```

Comment:
I guess I don't understand how this fixes it... explanation?

Reply:
I just added a note in 518509c.  Unfurling the URL there for convenience:

> Forcing an axis to be categorical
If you pass string values for the x or y parameter, plotly will automatically set the corresponding axis type to category, with the exception of string of numbers, in which case the axis is linear. It is however possible to force the axis type by setting explicitely xaxis_type to be category.

There's oddly no explanation of this in the Plotly JavaScript docs, but a similar generic API applies for all languages they support.

Reply:
Great - thanks!